eloquentjs/20_2_directory_creation/src/main.ts

```diff
@@ -0,0 +1,57 @@
+import { createServer, OutgoingMessage } from "node:http";
+import { GET } from "./modules/get";
+import { PUT } from "./modules/put";
+import { MKCOL } from "./modules/mkcol";
+import { DELETE } from "./modules/delete";
+
+interface ServerResponse {
+    status: number,
+    body?: any,
+    type?: string
+}
+
+interface ServerMethods {
+    GET: Function;
+    PUT: Function;
+    MKCOL: Function;
+    DELETE: Function;
+}
+
+type Method = keyof ServerMethods;
+
+const methods = { GET, PUT, MKCOL, DELETE }
```

Comment:
This typing approach feels somewhat overwrought to me. Why not just use a literal of the methods without the interface?